client/libra-dev/src/error.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use once_cell::sync::Lazy;
+use std::borrow::{Borrow, BorrowMut};
+use std::cmp::min;
+use std::os::raw::c_char;
+
+const MAX_ERROR_LENGTH: usize = 1024;
+
+static mut LAST_ERROR: Lazy<[u8; MAX_ERROR_LENGTH]> = Lazy::new(|| [0u8; MAX_ERROR_LENGTH]);
```

Comment:
thread_loca doesn’t work here because some language c bindings are not always handled by the same thread that runs that language code. 

Good point on multi threading issue, I think we just need an mutex/atomic wrapper outside of Lazy. 

We decided to use lazy because for error message it is very curbsom to alloc/free, so we use an global allocated static buffer and just Return const pointer to it. This has reenerancy issues but none of our api has reentrancy.